pkg/solana/logpoller/loader_test.go

```diff
@@ -0,0 +1,340 @@
+	blockHeight := uint64(21)
+
+	client.EXPECT().GetLatestBlockhash(mock.Anything, rpc.CommitmentFinalized).Return(&rpc.GetLatestBlockhashResult{
+		RPCContext: rpc.RPCContext{
+			Context: rpc.Context{
+				Slot: slot,
+			},
+		},
+	}, nil)
+
+	client.EXPECT().GetBlocks(mock.Anything, uint64(1), mock.MatchedBy(func(val *uint64) bool {
+		return val != nil && *val == slot
+	}), mock.Anything).Return(rpc.BlocksResult{slot}, nil)
+
+	client.EXPECT().GetBlockWithOpts(mock.Anything, slot, mock.Anything).Return(&rpc.GetBlockResult{
+		Transactions: []rpc.TransactionWithMeta{
+			{
+				Meta: &rpc.TransactionMeta{
+					LogMessages: messages,
+				},
+			},
+		},
+		Signatures:  []solana.Signature{sig},
+		BlockHeight: &blockHeight,
+	}, nil).Twice()
+
+	tests.AssertEventually(t, func() bool {
+		return parser.Called()
+	})
+
+	client.AssertExpectations(t)
+}
+
+func TestEncodedLogCollector_BackfillForAddress(t *testing.T) {
+	client := new(mocks.RPCClient)
+	parser := new(testParser)
+	ctx := tests.Context(t)
+
+	collector := logpoller.NewEncodedLogCollector(client, parser, logger.Nop())
+
+	require.NoError(t, collector.Start(ctx))
+	t.Cleanup(func() {
+		require.NoError(t, collector.Close())
+	})
+
+	pubKey := solana.PublicKey{2, 1, 4, 2}
+	slots := []uint64{42, 43, 44}
+	blockHeights := []uint64{21, 22, 23}
+	sig := solana.Signature{2, 1, 4, 2}
+
+	client.EXPECT().GetSignaturesForAddressWithOpts(mock.Anything, pubKey, mock.Anything).Return([]*rpc.TransactionSignature{
+		{Slot: slots[0]},
+		{Slot: slots[1]},
+		{Slot: slots[2]},
+	}, nil)
+
+	for idx := range len(slots) {
+		client.EXPECT().GetBlockWithOpts(mock.Anything, slots[idx], mock.Anything).Return(&rpc.GetBlockResult{
+			Transactions: []rpc.TransactionWithMeta{
+				{
+					Meta: &rpc.TransactionMeta{
+						LogMessages: messages,
+					},
+				},
+				{
+					Meta: &rpc.TransactionMeta{
+						LogMessages: messages,
+					},
+				},
+			},
+			Signatures:  []solana.Signature{sig, sig},
+			BlockHeight: &blockHeights[idx],
+		}, nil).Twice()
+	}
+
+	assert.NoError(t, collector.BackfillForAddress(ctx, pubKey.String(), 42))
+
+	tests.AssertEventually(t, func() bool {
+		return parser.Count() == 6
+	})
+
+	client.AssertExpectations(t)
+}
+
+func BenchmarkEncodedLogCollector(b *testing.B) {
+	ctx := tests.Context(b)
+
+	ticker := time.NewTimer(500 * time.Millisecond)
+	defer ticker.Stop()
+
+	parser := new(testParser)
+	blockProducer := &testBlockProducer{
+		b:         b,
+		nextSlot:  10,
+		blockSigs: make(map[uint64][]solana.Signature),
+		sigs:      make(map[string]bool),
+	}
+
+	collector := logpoller.NewEncodedLogCollector(blockProducer, parser, logger.Nop())
+
+	require.NoError(b, collector.Start(ctx))
+	b.Cleanup(func() {
+		require.NoError(b, collector.Close())
+	})
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+BenchLoop:
+	for i := 0; i < b.N; i++ {
+		select {
+		case <-ticker.C:
+			blockProducer.incrementSlot()
+		case <-ctx.Done():
+			continue BenchLoop
```

Comment:
Should this be break rather than continue? No point in continuing to loop over 0... b.N after the context is closed is there?

Reply:
I don't really think it matters because the context comes from the time the function is expected to run, but so is `b.N`. I'll make it a break statement instead just for better clarity.